pydantic/_internal/_generate_schema.py

```diff
@@ -1891,20 +1891,21 @@ def _unsubstituted_typevar_schema(self, typevar: typing.TypeVar) -> core_schema.
 
         bound = typevar.__bound__
         constraints = typevar.__constraints__
+        typevar_default_value = getattr(typevar, '__default__', None)
 
         try:
             typevar_has_default = typevar.has_default()  # type: ignore
         except AttributeError:
             # could still have a default if it's an old version of typing_extensions.TypeVar
-            typevar_has_default = getattr(typevar, '__default__', None) is not None
+            typevar_has_default = typevar_default_value is not None
```

Comment:
We can probably remove the `typevar_has_default` logic here and just go with the above